core/src/main/java/org/infinispan/health/HealthStatus.java

```diff
@@ -0,0 +1,24 @@
+package org.infinispan.health;
+
+/**
+ * General Health status.
+ *
+ * @author Sebastian Łaskawiec
+ * @since 9.0
+ */
+public enum HealthStatus {
+    /**
+     * Given entity is unhealthy.
+     */
+    RED,
+
+    /**
+     * Given entity is healthy.
+     */
+    GREEN,
+
+    /**
+     * Given entity is healthy but a rebalance is in progress.
+     */
+    YELLOW
```

Comment:
The review process is here to catch things at any time. I didn't think much about it at the time, but I do agree with @danberindei , although I'd probably change the middle state to something more specific than YELLOW but also less specific than REBALANCING (BUSY / REPAIRING, IN_MAINTENANCE ?), since we might want to indicate other situations where the cluster is best left (temporarily) alone.